src/ucp/core/ucp_context.c

```diff
@@ -668,7 +668,8 @@ static void ucp_add_tl_resource_if_enabled(ucp_context_h context, ucp_tl_md_t *m
 
     if (ucp_is_resource_enabled(resource, config, &rsc_flags, dev_cfg_masks,
                                 tl_cfg_mask)) {
-        if (resource->sys_device >= UCP_MAX_SYS_DEVICES) {
+        if ((resource->sys_device != UCS_SYS_DEVICE_ID_UNKNOWN) &&
```

Comment:
but would not it be still ignored if the device is `unknown`?

Reply:
pls see updated PR desc